src/game.py

```diff
@@ -0,0 +1,284 @@
+        return  -- Array of tuples [(Position, Value)], the number of array items depends of number gasses found
+        """
+        while True:
+            stuck, guess, count = True, None, 0
+            allowed, needed = self.calculate_bits(board)
+            try:
+                board, allowed, stuck, guess, count = self.move_on_board(board, allowed, stuck, guess, count)
+            except:
+                pass
+            if not stuck:
+                allowed, needed = self.calculate_bits(board)
+            board, stuck, guess, count = self.move_on_board_for_axis(needed, allowed, board, guess, count, stuck)
+            if stuck:
+                return self.get_is_stuck(guess)
+
+    def get_is_stuck(self, guess):
+        """Ask about guess value
+
+        Keyword arguments:
+        guess -- inherited from deduce function
+        return  -- return an integer from 1 to 9
+        """
+        if guess is not None:
+            random.shuffle(guess)
+        return guess
+
+    def move_on_board_for_axis(self, needed, allowed, board, guess, count, stuck):
+        """Move on board for each axis and guess needed and allowed values
+
+        Keyword arguments:
+        needed -- Array of numbers needed to resolve puzzle
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
+        guess -- Array of random numbers from 1 to 9
+        count -- Count how many times pick the best option of guesses
+        stuck -- Verify if game stuck returning true or false
+        return  -- Tuple of (board, stuck, guess, count)
+        """
+        for axis in xrange(3):
+            try:
+                board, stuck, guess, count = self.move_on_x_position(needed, allowed, board, guess, count, axis, stuck)
+            except:
+                pass
+        return board, stuck, guess, count
+
+    def move_on_x_position(self, needed, allowed, board, guess, count, axis, stuck):
+        """Move on axis position for each axis x  and guess needed and allowed values
+
+        Keyword arguments:
+        needed -- Array of numbers needed to resolve puzzle
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
+        guess -- Array of random numbers from 1 to 9
+        count -- Count how many times pick the best option of guesses
+        stuck -- Verify if game stuck returning true or false
+        axis -- Number for each iteration of move_on_board_for_axis function
+        return  -- Tuple of (board, stuck, guess, count)
+        """
+        for x in xrange(9):
+            numbers = self.list_bits(needed[axis * 9 + x])
+            result = self.move_for_numbers(allowed, board, guess, count, axis, x, numbers, stuck)
+            zero, board, stuck, guess, count = result
+            if zero == 0:
+                return []
+        return board, stuck, guess, count
```

Comment:
be consistent on your return values if for the general case you have a tuple for the other case you should return an empty tuple or none and not an array